sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/PropagateEmptyRelation.scala

```diff
@@ -108,5 +108,27 @@ object PropagateEmptyRelation extends Rule[LogicalPlan] with PredicateHelper wit
       case Generate(_: Explode, _, _, _, _, _) => empty(p)
       case _ => p
     }
+
+    // Nodes below GlobalLimit or LocalLimit can be pruned if the limit value is zero (0).
+    // Any subtree in the logical plan that has GlobalLimit 0 or LocalLimit 0 as its root is
+    // semantically equivalent to an empty relation.
+    //
+    // In such cases, the effects of Limit 0 can be propagated through the Logical Plan by replacing
+    // the (Global/Local) Limit subtree with an empty LocalRelation, thereby pruning the subtree
+    // below and triggering other optimization rules of PropagateEmptyRelation to propagate the
+    // changes up the Logical Plan.
+    //
+    // Replace Global Limit 0 nodes with empty Local Relation
+    case p @ GlobalLimit(IntegerLiteral(limit), _) if limit == 0 =>
```

Comment:
it's better to do a new rule, to make it more modular.

Reply:
+1, we can also create a new Batch for the new rule, since it requires only one-time transformation.

Reply:
@rxin @gengliangwang Hi! 
We are relying on the other case rules in `PropagateEmptyRelation` to propagate the effects of substituting Limit 0 with empty LocalRelation through the Logical Plan - such as in case of joins, project, etc. 
So, should we handle this case also part of this rule, and then propagation of empty relation happens only as part of this rule? Or do you recommend doing it as a separate rule?

Reply:
The new rule is not related to "propagate".  We can create a separate rule and batch , e.g. 
```
    Batch("OptimizeLimitZero", Once, OptimizeLimitZero) :+
    Batch("LocalRelation", fixedPoint,
      ConvertToLocalRelation,
      PropagateEmptyRelation
   )
```

Reply:
@rxin @gengliangwang @srowen I have made the recommended changes. Please have a look. Thanks!